internal/services/subscription/subscription_resource.go

```diff
@@ -273,15 +270,15 @@ func resourceSubscriptionUpdate(d *pluginsdk.ResourceData, meta interface{}) err
 		displayName := subscriptionAlias.SubscriptionName{
 			SubscriptionName: utils.String(d.Get("subscription_name").(string)),
 		}
-		if _, err := aliasClient.SubscriptionRename(ctx, *subscriptionId, displayName); err != nil {
-			return fmt.Errorf("could not update Display Name of Subscription %q: %+v", *subscriptionId, err)
+		if _, err := aliasClient.SubscriptionRename(ctx, subscriptionId, displayName); err != nil {
+			return fmt.Errorf("could not update Display Name of Subscription %q: %+v", subscriptionId, err)
 		}
 	}
 
 	if d.HasChange("tags") {
 		tagsClient := meta.(*clients.Client).Resource.TagsClientForSubscription(subscriptionId.ID())
 		t := tags.Expand(d.Get("tags").(map[string]interface{}))
-		scope := fmt.Sprintf("subscriptions/%s", *subscriptionId)
+		scope := fmt.Sprintf("subscriptions/%s", *resp.Model.Properties.SubscriptionId)
```

Comment:
We should just use the ID from `subscriptionId.ID()`
```suggestion
		scope := fmt.Sprintf("subscriptions/%s", subscriptionId.ID())
```

Reply:
subscriptionId.ID() returns "/subscriptions/00000000-0000-0000-0000-000000000000" which will cause [problem commented](https://github.com/hashicorp/terraform-provider-azurerm/pull/21255#discussion_r1155692458) by Tom.

Reply:
The URL sent to the Azure service will look like https://management.azure.com//subscriptions/00000000-0000-0000-0000-000000000000//...., which has `//` between `management.azure.com` and `subscriptions`. Though it can be accepted by Azure Service, maybe we should still pass a normal URL like https://management.azure.com/subscriptions/... to the backend to prevent any incompatibility problem. WDYT?

Reply:
FWIW `//` and `/` are both valid URIs (and supported by the Resource Manager API) - however `hashicorp/go-azure-sdk` should be accounting for this, can you confirm the HTTP Request and Response here?

Reply:
I have updated the code to use **subscriptionId.ID()**. The following logs are for tags-updating scenario with this change. Both request and response have `//`. Kindly take another review. 😃

**AzureRM Request:** 
PUT //subscriptions/00000000-0000-0000-0000-000000000000/providers/Microsoft.Resources/tags/default?api-version=2020-06-01 HTTP/1.1
{"properties":{"tags":{"key":"value2"}}}

**AzureRM Response** for https://management.azure.com//subscriptions/00000000-0000-0000-0000-000000000000/providers/Microsoft.Resources/tags/default?api-version=2020-06-01: 
HTTP/2.0 200 OK
{"id":"/subscriptions/00000000-0000-0000-0000-000000000000/providers/Microsoft.Resources/tags/default","name":"default","type":"Microsoft.Resources/tags","properties":{"tags":{"key":"value2"}}}

Reply:
Cool~, then I will resolve this comment.